js/column.js

```diff
@@ -330,20 +330,23 @@ ReferenceColumn.prototype = {
      */
     get display() {
         if (this._display_cached === undefined) {
-            var res = {};
-            if (res._simple) {
-                res = this._baseCols[0].getDisplay(this._context);
+            var res = {}, colDisplay = {}, sourceDisplay = {};
+            if (this._simple) {
+                colDisplay = this._baseCols[0].getDisplay(this._context);
             }
 
             // attach display defined on the source
             if (this.sourceObject && this.sourceObject.display) {
                 var displ = this.sourceObject.display;
                 if (typeof displ.markdown_pattern === "string") {
-                    res.sourceMarkdownPattern = displ.markdown_pattern;
-                    res.sourceTemplateEngine = displ.template_engine;
+                    sourceDisplay.sourceMarkdownPattern = displ.markdown_pattern;
+                    sourceDisplay.sourceTemplateEngine = displ.template_engine;
                 }
             }
 
+            // I'm using assign to avoid changing the original colDisplay
+            // it was causing issues and some columns were using other column's display properties
+            Object.assign(res, colDisplay, sourceDisplay);
```

Comment:
Correct. Instead of changing the original `colDisplay` and `sourceDisplay` objects, it will copy all the attributes of these two objects into the res (by value and not by reference).